plugins/logs/encoder.go

```diff
@@ -40,7 +41,15 @@ func (enc *chunkEncoder) Write(bs []byte) (result []byte, err error) {
 		if err := enc.writeClose(); err != nil {
 			return nil, err
 		}
+
 		result = enc.reset()
+		// add the chunked data to the buffer
+		if result != nil {
+			dropped := buffer.Push(result)
```

Comment:
I think it's better if this logic is kept inside the plugin instead of being moved into here. As is the encoder is calling printf (which we don't want to do). I don't see reason why the buffer append has to happen here.

Reply:
Done